### DISCUSSION TITLE
Model classes for InterceptorRequest and InterceptorResponse?

### DISCUSSION THREAD

**Discussion by ggallen** [fabric8io/kubernetes-client#5319] [Q&A]
Are there any model classes generated for Tekton Triggers InterceptorRequest and InterceptorResponse?

https://pkg.go.dev/github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1#InterceptorRequest
https://pkg.go.dev/github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1#InterceptorResponse

**Comment by rohanKanojia** [comment]
Are these model classes CRDs? How do you plan to use these? At the moment our tekton model only seems to expose CRDs

**Reply by ggallen** [reply]
No. I don't think they're CRDs. They are essentially the interface for calling an `Interceptor`.

My goal here is to write my own `Interceptor` (see https://github.com/fabric8io/kubernetes-client/discussions/5318).
